pubsub.go

```diff
@@ -22,6 +23,20 @@ import (
 
 var log = logging.Logger("pubsub-valuestore")
 
+type watcher struct {
+	ctx context.Context
+	out chan<- []byte
+}
+
+type watchGroup struct {
+	closing chan struct{}
+
+	register   chan watcher
+	unregister chan []chan<- []byte
```

Comment:
I'd rather use a struct for these, even if it's an anonymous one (to help newcomers to this code).

(assuming we need multiple channels)